src/main/java/com/mraof/minestuck/computer/editmode/EditToolDrag.java

```diff
@@ -0,0 +1,376 @@
+	}
+	
+	public static boolean canEditRecycle(Player player)
+	{
+		return (ClientEditHandler.isActive() && !isMultiblock(player));
+	}
+	
+	private static boolean isBlockDeployable(Player player)
+	{
+		ItemStack stack	= player.getMainHandItem().isEmpty() ? player.getOffhandItem() : player.getMainHandItem();
+	
+		return ClientDeployList.getEntry(stack) != null && stack.getItem() instanceof BlockItem && ((BlockItem) stack.getItem()).getBlock() instanceof MachineBlock;
+	}
+
+	private static boolean isMultiblock(Player player)
+	{
+		BlockPos blockLookingAt = getPlayerPOVHitResult(player.getLevel(), player).getBlockPos();
+		
+		return (player.getLevel().getBlockState(blockLookingAt) != null && (player.getLevel().getBlockState(blockLookingAt).getBlock() instanceof MultiMachineBlock));
+	}
+	
+	//based on the Item class function of the same name
+	private static BlockHitResult getPlayerPOVHitResult(Level level, Player playerEntity)
+	{
+		float xRot = playerEntity.getXRot();
+		float yRot = playerEntity.getYRot();
+		Vec3 eyeVec = playerEntity.getEyePosition(1.0F);
+		float f2 = Mth.cos(-yRot * ((float) Math.PI / 180F) - (float) Math.PI);
+		float f3 = Mth.sin(-yRot * ((float) Math.PI / 180F) - (float) Math.PI);
+		float f4 = -Mth.cos(-xRot * ((float) Math.PI / 180F));
+		float yComponent = Mth.sin(-xRot * ((float) Math.PI / 180F));
+		float xComponent = f3 * f4;
+		float zComponent = f2 * f4;
+		double reachDistance = playerEntity.getAttribute(ForgeMod.REACH_DISTANCE.get()).getValue();
+		Vec3 endVec = eyeVec.add((double) xComponent * reachDistance, (double) yComponent * reachDistance, (double) zComponent * reachDistance);
+		return level.clip(new ClipContext(eyeVec, endVec, ClipContext.Block.COLLIDER, ClipContext.Fluid.NONE, playerEntity));
+	}
+	
+	public static void updateEditToolsServer(ServerPlayer player, boolean isDragging, BlockPos pos1, BlockPos pos2)
+	{
+		IEditTools cap = player.getCapability(MSCapabilities.EDIT_TOOLS_CAPABILITY, null).orElse(new EditTools());
+		cap.setEditDragging(isDragging);
+		cap.setEditPos1(pos1);
+		cap.setEditPos2(pos2);
+		
+		boolean signX = pos1.getX() < pos2.getX();
+		boolean signY = pos1.getY() > pos2.getY();
+		boolean signZ = pos1.getZ() < pos2.getZ();
+		
+		double posX = pos2.getX() + (signX ? 1 : 0);
+		double posY = pos2.getY() + (signY ? 0 : 1);
+		double posZ = pos2.getZ() + (signZ ? 1 : 0);
+		boolean flipCursor = signY;
+		
+		float cursorLean = 0f;
+		if (signX && !signZ)
+			cursorLean = 360.0f; //+X -Z = 0/360
+		if (signX && signZ)
+			cursorLean = 90.0f; //+X +Z = 90
+		if (!signX && signZ)
+			cursorLean = 180.0f; //-X +Z = 180
+		if (!signX && !signZ)
+			cursorLean = 270.0f; //-X -Z = 270
+		
+		if(cap.getEditCursorID() == null)
+		{
+			cap.setEditCursorID(createCursorEntity(player, new Vec3(posX,posY,posZ), cursorLean, flipCursor));
+		}
+		else
+		{
+			updateCursorEntity(player, new Vec3(posX,posY,posZ), cursorLean, flipCursor, cap.getEditCursorID());
+		}
+	}
```

Comment:
Not questioning the math involved, but could you give a small explanatory javadoc comment?

Reply:
Added more comments in the latest commit. Let me know if it isn't clear enough.

Reply:
Looks good to me!